diff --git a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppQualifiedId.cs b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppQualifiedId.cs
index 9b66f6a3523daa7d792677df507fc050f1e65a42..f28d51e260b59d6d291cfc30b30a6ac6e2a11694 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppQualifiedId.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppQualifiedId.cs
@@ -90,6 +90,10 @@ public struct CppQualifiedId(string name)
 
 	public static readonly CppQualifiedId Rs__func__ = new CppQualifiedId("__func__");
 
+	public static readonly CppQualifiedId PLATFORM_NAME = new CppQualifiedId("Platform"); // @MOD C++/CX
+	public static readonly CppQualifiedId DELEGATE_NAME = new CppQualifiedId("Delegate"); // @MOD C++/CX
+	public static readonly CppQualifiedId WRITE_ONLY_ARRAY_NAME = new CppQualifiedId("WriteOnlyArray"); // @MOD C++/CX
+
 	public string Name { get; } = name;
 
 	public bool Equals(CppQualifiedNamePart rhs)
