diff --git a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppNodeTypeCounter.cs b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppNodeTypeCounter.cs
index 3f787daab4d5809096cdbdf7f514d28df85a7b85..f5e4ac21bf94e3f45d42cbf24f1948ee78870bb2 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppNodeTypeCounter.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppNodeTypeCounter.cs
@@ -4,7 +4,7 @@ namespace JetBrains.ReSharper.Psi.Cpp.Parsing;
 
 public class CppNodeTypeCounter
 {
-	private static int ourTokenTypeCounter = 587;
+	private static int ourTokenTypeCounter = (int)CppTokenKind._NUMBER_OF_ELEMENTS_; // @MOD C++/CX
 
 	public static int IncIndex()
 	{
