diff --git a/JetBrains.ReSharper.Psi.Cpp.Expressions/CppExpressionsUtil.cs b/JetBrains.ReSharper.Psi.Cpp.Expressions/CppExpressionsUtil.cs
index 87a8dff16b7c63c28497819255a095da11e9d11e..a6f4f407e59ff0c0808c18f9ec255f30b90455b7 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Expressions/CppExpressionsUtil.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Expressions/CppExpressionsUtil.cs
@@ -501,13 +501,34 @@ public static class CppExpressionsUtil
 		{
 			return UnaryRequiresOverloading(arg, tc.Factory);
 		}
-		if (arg is ICppResolvedQualifiedReferenceExpression cppResolvedQualifiedReferenceExpression && CheckQualifiedReferenceForMemberType(cppResolvedQualifiedReferenceExpression, out var _, in tc))
+		if (CheckQualifiedReferenceForMemberType(arg, out var _, in tc)) // @MOD Fix RSCPP-15665
 		{
 			return false;
 		}
 		return UnaryRequiresOverloading(arg, tc.Factory);
 	}
 
+	// @MOD Fix RSCPP-15665
+	internal static bool CheckQualifiedReferenceForMemberType(ICppResolvedExpression expression, out CppQualType memberT, in CppTypeContext tc)
+	{
+		if (expression is ICppParenExpression)
+		{
+			if (tc.Factory.GetDialect() is CppMSVCDialect and not CppClangOnWindowsToolset or CustomLanguageDialect { CompilerKind: Rider.Model.RadCompilerKind.MSVC })
+			{
+				while (expression is ICppParenExpression paren)
+				{
+					expression = (ICppResolvedExpression)paren.GetInnerExpression();
+				}
+			}
+		}
+		if (expression is ICppResolvedQualifiedReferenceExpression reference)
+		{
+			return CheckQualifiedReferenceForMemberType(reference, out memberT, in tc);
+		}
+		memberT = CppQualType.EmptyType();
+		return false;
+	}
+
 	public static bool CheckQualifiedReferenceForMemberType(ICppResolvedQualifiedReferenceExpression @ref, out CppQualType memberT, in CppTypeContext tc)
 	{
 		ICppDeclaratorResolveEntity declaratorIfMember = GetDeclaratorIfMember(@ref, in tc);
@@ -982,6 +1003,11 @@ public static class CppExpressionsUtil
 		{
 			CppTypeAndCategory typeAndCategory = resolvedLeftArgument.GetTypeAndCategory(tf);
 			typeAndCategory.Type = typeAndCategory.Type.GetPointeeIfPointer().ClearRefs();
+			// @MOD C++/CX: Projected enum members operate on the value regardless of top level CV qualification
+			if (tf.GetDialect().LanguageKind == CppLanguageKind.CppCx && typeAndCategory.Type.IsCliEnum())
+			{
+				typeAndCategory.Type = typeAndCategory.Type.CVUnqualify();
+			}
 			return typeAndCategory;
 		}
 		CppResolveResult lookupScopeResolveResult = expr.GetLookupScopeResolveResult();
