diff --git a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppDeclarationBuilderUtil.cs b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppDeclarationBuilderUtil.cs
index e9b886137ef6a76dc006c88d71e7d24cb9f62c5b..2502e7155c5b95d09fc055e29bd662d9524b10e6 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppDeclarationBuilderUtil.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppDeclarationBuilderUtil.cs
@@ -100,9 +100,15 @@ public static class CppDeclarationBuilderUtil
 		opParenDeclaratorPack.AddGroupedDeclarator(CppGroupedFunctionDeclaratorResolveEntity.Create(cppParsedFunctionDeclaratorResolveEntity));
 		CppLocationAnchor anchor2 = resolveCache.IncrementCurrentAnchor();
 		CppDeclaratorResolveEntityPack orCreateConstructors = cppMutableClassResolveEntity.GetOrCreateConstructors(anchor2);
+		bool isCx = resolveCache.TypeFactory.GetDialect().LanguageKind == Language.CppLanguageKind.CppCx; // @MOD C++/CX
+		if (!isCx) // @MOD C++/CX
 		orCreateConstructors.AddDeclarator(new CppCxxCliDelegateConstructorResolveEntity(cppMutableClassResolveEntity, orCreateConstructors.Name, CppCxxCliDelegateConstructorResolveEntity.CtorType.GLOBAL_FUNC_CLRCALL));
 		orCreateConstructors.AddDeclarator(new CppCxxCliDelegateConstructorResolveEntity(cppMutableClassResolveEntity, orCreateConstructors.Name, CppCxxCliDelegateConstructorResolveEntity.CtorType.GLOBAL_FUNC_CDECL));
 		orCreateConstructors.AddDeclarator(new CppCxxCliDelegateConstructorResolveEntity(cppMutableClassResolveEntity, orCreateConstructors.Name, CppCxxCliDelegateConstructorResolveEntity.CtorType.GLOBAL_FUNC_STDCALL));
+		if (isCx) // @MOD C++/CX
+		{
+			orCreateConstructors.AddDeclarator(new CppCxxCliDelegateConstructorResolveEntity(cppMutableClassResolveEntity, orCreateConstructors.Name, CppCxxCliDelegateConstructorResolveEntity.CtorType.CX_LAMBDA));
+		}
 		if (replacedDelegateType.GetFunctionParametersCount() > 0)
 		{
 			CppQualType functionParameterType = cppParsedFunctionDeclaratorResolveEntity.GetCppType(resolveCache.TypeFactory).GetFunctionParameterType(0);
