diff --git a/JetBrains.ReSharper.Psi.Cpp.Expressions/CppSingleArgumentTypeTraitKindUtil.cs b/JetBrains.ReSharper.Psi.Cpp.Expressions/CppSingleArgumentTypeTraitKindUtil.cs
index 467aefce2a490fc4f2faa4724937db5bc6bc2e92..8e280ff15332b99fd0e7d2b80f9d59e0b9294310 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Expressions/CppSingleArgumentTypeTraitKindUtil.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Expressions/CppSingleArgumentTypeTraitKindUtil.cs
@@ -79,6 +79,12 @@ public static class CppSingleArgumentTypeTraitKindUtil
 		Add(CppTokenNodeTypes.__IS_INTEGRAL, CppSingleArgumentTypeTraitKind.IS_INTEGRAL);
 		Add(CppTokenNodeTypes.__IS_FLOATING_POINT, CppSingleArgumentTypeTraitKind.IS_FLOATING_POINT);
 		Add(CppTokenNodeTypes.__IS_IMPLICIT_LIFETIME, CppSingleArgumentTypeTraitKind.IS_IMPLICIT_LIFETIME);
+
+		// @MOD C++/CX
+		Add(CppTokenNodeTypes.__IS_WIN_CLASS, CppSingleArgumentTypeTraitKind.IS_WIN_CLASS);
+		Add(CppTokenNodeTypes.__IS_WIN_INTERFACE, CppSingleArgumentTypeTraitKind.IS_WIN_INTERFACE);
+		Add(CppTokenNodeTypes.__IS_WINRT_AGILE, CppSingleArgumentTypeTraitKind.IS_WINRT_AGILE);
+		Add(CppTokenNodeTypes.__IS_VALID_WINRT_TYPE, CppSingleArgumentTypeTraitKind.IS_VALID_WINRT_TYPE);
 		static void Add(TokenNodeType token, CppSingleArgumentTypeTraitKind kind)
 		{
 			ourTokenToKind.Add(token, kind);
