diff --git a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenNodeType.cs b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenNodeType.cs
index 4055012055e94427a89dd066f96c628d96f25756..4f096eac7f35e4f1b470dd32ef2ad3d73ac64f88 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenNodeType.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenNodeType.cs
@@ -7,7 +7,7 @@ namespace JetBrains.ReSharper.Psi.Cpp.Parsing;
 
 public abstract class CppTokenNodeType : TokenNodeType
 {
-	private static readonly CppTokenNodeType[] ourTokenByIndex = new CppTokenNodeType[587];
+	private static readonly CppTokenNodeType[] ourTokenByIndex = new CppTokenNodeType[(int)CppTokenKind._NUMBER_OF_ELEMENTS_]; // @MOD C++/CX
 
 	public override bool IsWhitespace
 	{
