diff --git a/JetBrains.ReSharper.Psi.Cpp.Caches/CppGlobalSymbolCache.cs b/JetBrains.ReSharper.Psi.Cpp.Caches/CppGlobalSymbolCache.cs
index 408bea641708dfc97a4811c71102fc04de049010..d12b1c08545a2fd4d8387a6654375432d6b11bad 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Caches/CppGlobalSymbolCache.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Caches/CppGlobalSymbolCache.cs
@@ -53,6 +53,9 @@ namespace JetBrains.ReSharper.Psi.Cpp.Caches;
 [PsiComponent(Instantiation.DemandAnyThreadUnsafe)]
 public sealed class CppGlobalSymbolCache : IStartupActivity
 {
+	public const int CACHE_FORMAT_VERSION_ORIGINAL = CACHE_FORMAT_VERSION; // @MOD C++/CX
+	public const int CACHE_FORMAT_VERSION_CX = CACHE_FORMAT_VERSION_ORIGINAL + 1; // @MOD C++/CX
+
 	public delegate void FileSymbolsRemoved(ICollection<CppFileLocation> files);
 
 	private class SourceFileState(LongSet symbolTables, int compilationPropertiesHashCode, int includePathsHashCode)
