diff --git a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppCxxCliClrPropertyResolveEntity.cs b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppCxxCliClrPropertyResolveEntity.cs
index 21966494575d75bd1a341db695e10eb0420e2e79..ee941bfd1dcc4b183de47af7936852003cc5f69d 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppCxxCliClrPropertyResolveEntity.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppCxxCliClrPropertyResolveEntity.cs
@@ -16,7 +16,7 @@ public sealed class CppCxxCliClrPropertyResolveEntity : CppCxxCliPropertyResolve
 	public IClrDeclaredElement ClrDeclaredElement => myTypeMember;
 
 	public CppCxxCliClrPropertyResolveEntity(ICppResolveEntityParentScope parent, IProperty prop, in CppClrTypeConverter cvt)
-		: base(parent, CppClrNameConverter.ConvertName(prop), CppLocationAnchor.GeneratedEntity(), cvt.ConvertType(prop.Type))
+		: base(parent, CppClrNameConverter.ConvertName(prop), CppLocationAnchor.GeneratedEntity(), cvt.ConvertMemberType(prop)) // @MOD C++/CX
 	{
 		myTypeMember = prop;
 		DeclarationSpecifiers = CppClrTypeConverter.CalculateDeclarationSpecifiers(prop);
@@ -30,7 +30,7 @@ public sealed class CppCxxCliClrPropertyResolveEntity : CppCxxCliPropertyResolve
 	}
 
 	public CppCxxCliClrPropertyResolveEntity(ICppResolveEntityParentScope parent, IEvent @event, in CppClrTypeConverter cvt)
-		: base(parent, CppClrNameConverter.ConvertName(@event), CppLocationAnchor.GeneratedEntity(), cvt.ConvertType(@event.Type))
+		: base(parent, CppClrNameConverter.ConvertName(@event), CppLocationAnchor.GeneratedEntity(), cvt.ConvertMemberType(@event)) // @MOD C++/CX
 	{
 		myTypeMember = @event;
 		DeclarationSpecifiers = CppClrTypeConverter.CalculateDeclarationSpecifiers(@event);
