diff --git a/JetBrains.ReSharper.Feature.Services.Cpp.csproj b/JetBrains.ReSharper.Feature.Services.Cpp.csproj
index f15ec25b189f44f8dba07e122a81a412f0b34e24..d645080be444379c217bb26cb42da22ed5f3ca98 100644
--- a/JetBrains.ReSharper.Feature.Services.Cpp.csproj
+++ b/JetBrains.ReSharper.Feature.Services.Cpp.csproj
@@ -87,6 +87,7 @@
     <Reference Include="JetBrains.Platform.Core">
       <HintPath>..\..\References\JetBrains.Platform.Core.dll</HintPath>
     </Reference>
+    <Reference Include="PresentationCore" /> <!--@MOD Fix compile; System.Windows.Markup.IAddChild-related errors-->
     <Reference Include="System.Core" />
     <Reference Include="JetBrains.Platform.Metadata">
       <HintPath>..\..\References\JetBrains.Platform.Metadata.dll</HintPath>
