diff --git a/JetBrains.ReSharper.Feature.Services.Cpp.csproj b/JetBrains.ReSharper.Feature.Services.Cpp.csproj
index d7b93564075bd19293f6cecf813de572e1c7e211..6f0a0bb321f9ebfe255c77f816270fe1ba4fe098 100644
--- a/JetBrains.ReSharper.Feature.Services.Cpp.csproj
+++ b/JetBrains.ReSharper.Feature.Services.Cpp.csproj
@@ -87,6 +87,7 @@
     <Reference Include="JetBrains.Platform.Core">
       <HintPath>..\..\References\JetBrains.Platform.Core.dll</HintPath>
     </Reference>
+    <Reference Include="PresentationCore" /> <!--@MOD Fix compile; System.Windows.Markup.IAddChild-related errors-->
     <Reference Include="System.Core" />
     <Reference Include="JetBrains.Platform.Metadata">
       <HintPath>..\..\References\JetBrains.Platform.Metadata.dll</HintPath>
