diff --git a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppDeclarationBuilderUtil.cs b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppDeclarationBuilderUtil.cs
index 9cbeaea2da306872ff8e7568b36a5ffdc2307107..c0c059a27a8a34ce68ec9e1059a933f2e2afe024 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppDeclarationBuilderUtil.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppDeclarationBuilderUtil.cs
@@ -101,9 +101,15 @@ public static class CppDeclarationBuilderUtil
 		opParenDeclaratorPack.AddGroupedDeclarator(CppGroupedFunctionDeclaratorResolveEntity.Create(cppParsedFunctionDeclaratorResolveEntity));
 		CppLocationAnchor anchor2 = resolveCache.IncrementCurrentAnchor();
 		CppDeclaratorResolveEntityPack orCreateConstructors = cppMutableClassResolveEntity.GetOrCreateConstructors(anchor2);
+		bool isCx = resolveCache.TypeFactory.GetDialect().LanguageKind == Language.CppLanguageKind.CppCx; // @MOD C++/CX
+		if (!isCx) // @MOD C++/CX
 		orCreateConstructors.AddDeclarator(new CppCxxCliDelegateConstructorResolveEntity(cppMutableClassResolveEntity, orCreateConstructors.Name, CppCxxCliDelegateConstructorResolveEntity.CtorType.GLOBAL_FUNC_CLRCALL));
 		orCreateConstructors.AddDeclarator(new CppCxxCliDelegateConstructorResolveEntity(cppMutableClassResolveEntity, orCreateConstructors.Name, CppCxxCliDelegateConstructorResolveEntity.CtorType.GLOBAL_FUNC_CDECL));
 		orCreateConstructors.AddDeclarator(new CppCxxCliDelegateConstructorResolveEntity(cppMutableClassResolveEntity, orCreateConstructors.Name, CppCxxCliDelegateConstructorResolveEntity.CtorType.GLOBAL_FUNC_STDCALL));
+		if (isCx) // @MOD C++/CX
+		{
+			orCreateConstructors.AddDeclarator(new CppCxxCliDelegateConstructorResolveEntity(cppMutableClassResolveEntity, orCreateConstructors.Name, CppCxxCliDelegateConstructorResolveEntity.CtorType.CX_LAMBDA));
+		}
 		if (replacedDelegateType.GetFunctionParametersCount() > 0)
 		{
 			CppQualType functionParameterType = cppParsedFunctionDeclaratorResolveEntity.GetCppType(resolveCache.TypeFactory).GetFunctionParameterType(0);
