diff --git a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenKind.cs b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenKind.cs
index ba3ff4d0b03276b01c0e8499079f38353b60ec37..e7dd665ce2a6d4dda8f5a9464a9a94700d6b404e 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenKind.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenKind.cs
@@ -2,6 +2,15 @@ namespace JetBrains.ReSharper.Psi.Cpp.Parsing;
 
 public enum CppTokenKind : short
 {
+	// @MOD C++/CX
+	CXXCX_PARTIAL_KEYWORD = _NUMBER_OF_ELEMENTS_,
+	__IS_WIN_CLASS,
+	__IS_WIN_INTERFACE,
+	__IS_WINRT_AGILE,
+	__IS_VALID_WINRT_TYPE,
+
+	_NUMBER_OF_ELEMENTS_CX_,
+
 	INVALID = -1,
 	EOF = 10,
 	WHITE_SPACE = 11,
