diff --git a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppNodeTypeCounter.cs b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppNodeTypeCounter.cs
index 3001bce152c2b8774010c51157edd6e446fe0fbb..54f86855906c88f5b86ad8350ec48241ad2ddab8 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppNodeTypeCounter.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppNodeTypeCounter.cs
@@ -4,7 +4,7 @@ namespace JetBrains.ReSharper.Psi.Cpp.Parsing;
 
 public class CppNodeTypeCounter
 {
-	private static int ourTokenTypeCounter = 620;
+	private static int ourTokenTypeCounter = (int)CppTokenKind._NUMBER_OF_ELEMENTS_CX_; // @MOD C++/CX
 
 	public static int IncIndex()
 	{
