diff --git a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenNodeType.cs b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenNodeType.cs
index fbed00e50580c7f83288d9e3726c99424966d976..53b977b4b7b938a852f794327d260ded2e80afa2 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenNodeType.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Parsing/CppTokenNodeType.cs
@@ -7,7 +7,7 @@ namespace JetBrains.ReSharper.Psi.Cpp.Parsing;
 
 public abstract class CppTokenNodeType : TokenNodeType
 {
-	private static readonly CppTokenNodeType[] ourTokenByIndex = new CppTokenNodeType[620];
+	private static readonly CppTokenNodeType[] ourTokenByIndex = new CppTokenNodeType[(int)CppTokenKind._NUMBER_OF_ELEMENTS_CX_]; // @MOD C++/CX
 
 	public override bool IsWhitespace
 	{
