diff --git a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppSymbolEqualsVisitor.cs b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppSymbolEqualsVisitor.cs
index fa834d4acafc96de7011132bf46154fc81c02ec4..9c45ec7ff6190b9d1fb9a7973232bfd3f8138ce0 100644
--- a/JetBrains.ReSharper.Psi.Cpp.Symbols/CppSymbolEqualsVisitor.cs
+++ b/JetBrains.ReSharper.Psi.Cpp.Symbols/CppSymbolEqualsVisitor.cs
@@ -26,6 +26,11 @@ public class CppSymbolEqualsVisitor : ICppSymbolVisitor<bool>
 		{
 			return false;
 		}
+		// @MOD C++/CX: changing partial to final changes class composition even with identical members.
+		if (lhs.IsCxPartial != cppClassSymbol.IsCxPartial)
+		{
+			return false;
+		}
 		if (lhs.Key != cppClassSymbol.Key || !AnchorDeltaAreEqual(lhs, cppClassSymbol) || !BaseDescriptionListsAreEqual(lhs.GetBases(), cppClassSymbol.GetBases()) || !TemplateParameterSymbolsListsAreEqual(lhs.ParamList(), cppClassSymbol.ParamList()) || !VisitDefinitionClassSymbol(lhs, cppClassSymbol) || lhs.IsExplicitInstantiation != cppClassSymbol.IsExplicitInstantiation || !Engine.AttributeListsAreEqual(lhs.Attributes(), cppClassSymbol.Attributes()))
 		{
 			return false;
@@ -48,7 +53,7 @@ public class CppSymbolEqualsVisitor : ICppSymbolVisitor<bool>
 
 	public bool Visit(CppDeserializedFromModuleClassSymbol lhs)
 	{
-		return VisitSymbol(lhs);
+		return lhs.IsCxPartial == CppClassSymbolUtil.IsCxPartial(Rhs as ICppClassOrDelegateSymbolOrModuleEntity) && VisitSymbol(lhs); // @MOD C++/CX
 	}
 
 	public virtual bool Visit(CppEnumWithBaseSymbol lhs)
